broadinstitute/single_cell_portal_core
# Adjust GitHub Actions for branch deletion (SCP-4389)

This fixes a papercut found where a hot-fix release branch would be automatically deleted after merging it to Master which is a problem since it was needed for merging to Development after. 

This PR splits our development and master merge actions and also suggests a new requirement for hot-fix branches
1. Merge to master will not delete a branch that starts with 'hot-fix' or 'hotfix' (also does not delete development or master)
2. Merge to development will delete any branch that is not master (or development)

The new requirement that I will add to the playbook if folks agree is that for a hot-fix you must name the branch starting with 'hot-fix' or 'hotfix' for the Actions to act appropriately. This practice appears to already be something many folks are doing informally, so this is more so just formalizing that practice. 


https://user-images.githubusercontent.com/54322292/218143345-edbdd445-b2a3-452f-b62b-b83a8db46276.mov



To test:
- Use this Repo: https://github.com/ehanna4/testing-actions
- I have made a variety of branches available there that include hotfix branches and non-hotfix branches
- Try out merging a hotfix branch to master and see that it doesn't delete
- Try out merging a hotfix branch to development and see that it does delete
- Try out merging development to master and see it does not delete
- etc

Alternatively you can take a look at the outputs from the already successful runs of the github actions in the Actions tab
